src/Microsoft.Diagnostics.Runtime/src/Utilities/PEImage/PEImage.cs

```diff
@@ -257,6 +259,77 @@ public int Read(int virtualAddress, Span<byte> dest)
             }
         }
 
+        /// <summary>
+        /// Returns the address of a module export symbol if found
+        /// </summary>
+        /// <param name="symbolName">symbol name (without the module name prepended)</param>
+        /// <param name="offset">symbol offset returned</param>
+        /// <returns>true if found</returns>
+        public bool TryGetExportSymbol(string symbolName, out ulong offset)
+        {
+            try
+            {
+                if (_exportDirectory is null)
+                {
+                    if (PEHeader is not null)
+                    {
+                        DirectoryEntry exportTableDirectory = PEHeader.ExportTableDirectory;
+                        if (exportTableDirectory.RelativeVirtualAddress != 0 && exportTableDirectory.Size != 0)
+                        {
+                            _exportDirectory = Read<IMAGE_EXPORT_DIRECTORY>(RvaToOffset(exportTableDirectory.RelativeVirtualAddress));
+                        }
+                    }
+                }
+                if (_exportDirectory is not null)
+                {
+                    IMAGE_EXPORT_DIRECTORY exportDirectory = _exportDirectory.Value;
+
+                    for (int nameIndex = 0; nameIndex < exportDirectory.NumberOfNames; nameIndex++)
+                    {
+                        int namePointerRVA = Read<int>(RvaToOffset(exportDirectory.AddressOfNames + (sizeof(uint) * nameIndex)));
+                        if (namePointerRVA != 0)
+                        {
+                            string name = ReadNullTerminatedAscii(namePointerRVA);
+                            if (name == symbolName)
+                            {
+                                ushort ordinalForNamedExport = Read<ushort>(RvaToOffset(exportDirectory.AddressOfNameOrdinals + (sizeof(ushort) * nameIndex)));
+                                int exportRVA = Read<int>(RvaToOffset(exportDirectory.AddressOfFunctions + (sizeof(uint) * ordinalForNamedExport)));
+                                offset = (uint)RvaToOffset(exportRVA);
+                                return true;
+                            }
+                        }
+                    }
+                }
+            }
+            catch (Exception ex) when (ex is IOException || ex is InvalidDataException)
+            {
+            }
+            offset = 0;
+            return false;
+        }
+
+        private string ReadNullTerminatedAscii(int rva)
```

Comment:
This should take a max length.  The default max length can be very high (like 4096 or 8192), but there should definitely be a cap to the number of bytes read since malformed or weird crash dumps could cause an incredibly, incredibly slow read until we OOM.